src/fsharp/NameResolution.fs

```diff
@@ -1538,14 +1538,18 @@ type TcResultsSinkImpl(g, ?source: string) =
     let formatStringCheckContext =
         lazy
             source |> Option.map (fun source ->
-                let source = source.Replace("\r\n", "\n").Replace("\r", "\n")
                 let positions =
-                    source.Split('\n')
-                    |> Seq.map (fun s -> String.length s + 1)
-                    |> Seq.scan (+) 0
-                    |> Seq.toArray
-                { NormalizedSource = source 
-                  LineEndPositions = positions })
+                    [|
+                        yield 0
+                        for i in 0..source.Length-1 do
+                            let c = source.[i]
+                            if c = '\r' && i + 1 < source.Length && source.[i+1] = '\n' then ()
+                            elif c = '\r' then yield i + 1
+                            if c = '\n' then yield i + 1
```

Comment:
Should not this be an `elif`? (It does not change the semantics but improves clarity.) @AviAvni

Reply:
I think it's fine. alter native would be to pattern match on c